source/game/field/obj/ObjectDokan.hh

```diff
@@ -1,10 +1,10 @@
 #pragma once
 
-#include "game/field/obj/ObjectBase.hh"
+#include "game/field/obj/ObjectCollidable.hh"
 
 namespace Field {
 
-class ObjectDokan : public ObjectBase {
+class ObjectDokan : public ObjectCollidable {
```

Comment:
I like that you have `ObjectNoImpl` marked `final`. I think for consistency and readability, this should also be final. Same with oil

Reply:
I'm not entirely sure those are final. `ObjectNoImpl` is designed to be final, as it's impossible for there to be a class that derives it.

Reply:
We can pretty easily confirm nothing derives from these object classes, right? I'd be surprised if they are not final